### PR TITLE
Adding preparePasteEdits method to check if smart copy/paste should be applied

### PR DESCRIPTION
**Pull request by navya9singh**
This pr adds a `preparePasteEdits` api to check if `pasteEdits` should be performed.  https://github.com/microsoft/TypeScript/issues/59881

One drawback here is, a case where the copied text is `export const a = 1;`, this would return true, because it checks if an identifier is exported in the current source file. This is needed, so that a case like 
```
export const a = 1;
[|const b = a;|]
```  
where `const b = a` is the copied text, can be true.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -2300,6 +2303,16 @@ export function createLanguageService(
         };
     }
 
+    function preparePasteEditsForFile(fileName: string, copiedTextRange: TextRange[]): boolean {
+        const sourceFile = syntaxTreeCache.getCurrentSourceFile(fileName);
+        bindSourceFile(sourceFile, getDefaultCompilerOptions());
```
Comment (sheetalkamat) [inline src/services/services.ts:2308]: You are not suppose to bind the file on syntax server. Also the options used here are wrong as they should be `host.getCompilationSettings()`

Comment (navya9singh) [reply src/services/services.ts:2308]: Without the binding, I don't get any `locals` or `symbol.exports` in the sourceFile. Is there a different way of doing this?

Comment (sheetalkamat) [reply src/services/services.ts:2308]: If thats the case you would need this command on "partial semantic" or "semantic server" instead

**Review by mjbvz — APPROVED** [review]
Protocol changes look good. Thanks for looking into this!  Will adopt it on the VS Code side once merged

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**